### PR TITLE
Add Getting Started for Devs

### PR DESCRIPTION
**Pull request by thisisdano**
- Adds Getting Started content for developers
- Incorporates Ruby work from https://github.com/uswds/uswds-site/pull/1318

Replaces https://github.com/uswds/uswds-site/pull/1318 and https://github.com/uswds/uswds-site/pull/1360

[Preview](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.app.cloud.gov/preview/uswds/uswds-site/dw-add-getting-started-for-devs/documentation/getting-started-for-developers/) →

### CODE REVIEW

File: _data/nav.yml [hunk 1]
```diff
@@ -75,14 +75,14 @@ Components:
...
-    href: /documentation/fundamentals/
+  - text: Getting started for developers
+    href: /documentation/getting-started-for-developers/
+  - text: Getting started for designers
+    href: /documentation/getting-started-for-designers/
   - href: /documentation/developers/
-  - href: /documentation/designers/
-  - href: /documentation/accessibility/
   - href: /documentation/settings/
+  - href: /maturity-model/
```
Comment (thisisdano) [inline _data/nav.yml:84]: Moves Maturity Model down in the side nav

File: _data/nav.yml [hunk 2]
```diff
@@ -91,6 +91,7 @@ How to use USWDS:
   - href: /documentation/implementations/
   - href: /documentation/showcase/
   - href: /documentation/sample-contract-language/
+  - href: /website-standards/
```
Comment (thisisdano) [inline _data/nav.yml:94]: Moves Website Standards down in the side nav

File: _data/nav.yml [hunk 3]
```diff
@@ -75,14 +75,14 @@ Components:
...
 How to use USWDS:
   - href: /how-to-use-uswds/
-  - href: /website-standards/
-  - href: /maturity-model/
-  - text: USWDS fundamentals and quickstart guide
-    href: /documentation/fundamentals/
+  - text: Getting started for developers
+    href: /documentation/getting-started-for-developers/
+  - text: Getting started for designers
+    href: /documentation/getting-started-for-designers/
```
Comment (thisisdano) [inline _data/nav.yml:81]: These replace the old "For developers" and "For designers" sections

File: _data/nav.yml [hunk 4]
```diff
@@ -75,14 +75,14 @@ Components:
...
   - href: /documentation/developers/
```
Comment (thisisdano) [inline _data/nav.yml:82]: ↑ This is the old developers section, now called "Documentation"

File: _includes/step-indicator.html [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        {% assign modifier = "" %}
+      {% endif %}
+      <li class="usa-step-indicator__segment usa-step-indicator__segment{{ modifier }}">
+        <span class="usa-step-indicator__segment-label"
+          >{{ step.name }}{% if step.status == "completed" %} <span class="usa-sr-only">completed</span>{% endif %}{% if step.status == "completed" %} <span class="usa-sr-only">not completed</span>{% endif %}</span
+        >
+      </li>
+    {% endfor %}
+  </ol>
+  <div class="usa-step-indicator__header">
```
Comment (thisisdano) [inline _includes/step-indicator.html:18]: Added a step indicator component for testing, but ended up not implementing it here

File: _plugins/jekyll_get.rb [hunk 6]
```diff
@@ -35,12 +35,12 @@ def load_json(site, d)
       if not File.exists?(path)
         FileUtils.mkpath File.dirname(path)
         print "Caching #{url} in #{path}...\n"
-        data = JSON.load(open(get_final_url(url)))
-        open(path, 'wb') do |file|
+        data = JSON.load(URI.open(get_final_url(url)))
+        URI.open(path, 'wb') do |file|
```
Comment (thisisdano) [inline _plugins/jekyll_get.rb:39]: This is the Ruby work from https://github.com/uswds/uswds-site/pull/1318

File: pages/documentation/developers.md [hunk 7]
```diff
@@ -1,7 +1,7 @@
 ---
 permalink: /documentation/developers/
 layout: styleguide
-title: For developers
+title: Documentation
```
Comment (thisisdano) [inline pages/documentation/developers.md:4]: The old developer docs. Next step: rewriting and rethinking.

File: pages/documentation/getting-started-designers/designers/overview.md [hunk 8]
```diff
@@ -1,11 +1,11 @@
...
-permalink: /documentation/designers/
+permalink: /documentation/getting-started-for-designers/
 layout: styleguide
-title: For designers
+title: Getting started for designers
 category: How to use USWDS
 lead: USWDS offers a flexible yet coherent visual style you can apply to any government website. Our visual style offers a clean, modern aesthetic that meets high standards of usability and accessibility while promoting credibility and trust.
 redirect_from:
-  - /getting-started/designers/
+  - /documentation/designers/
```
Comment (thisisdano) [inline pages/documentation/getting-started-designers/designers/overview.md:8]: Redirects from the old designers page

File: pages/documentation/getting-started-developers/overview.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+title: Getting started for developers
+lead: Get started using the design system by installing, compiling, and customizing USWDS code
+category: How to use USWDS
+type: docs
+subnav:
+  - href: /documentation/getting-started/developers/phase-one-install/
+  - href: /documentation/getting-started/developers/phase-two-compile/
+  - href: /documentation/getting-started/developers/phase-three-customize/
+redirect_from:
+  - /documentation/fundamentals/
```
Comment (thisisdano) [inline pages/documentation/getting-started-developers/overview.md:13]: Redirects from the old fundamentals and quickstart page

File: pages/documentation/overview.md [hunk 10]
```diff
@@ -15,15 +15,15 @@ redirect_from:
   <div class="tablet:grid-col display-flex flex-align-stretch">
     <div class="site-docs-card-link">
       <h3 class="font-lang-lg margin-0">
-        <a href="{{ site.baseurl }}/getting-started/developers/" class="text-no-underline text-primary hover:text-underline block-link">Developers</a>
+        <a href="{{ site.baseurl }}/documentation/getting-started-for-developers/" class="text-no-underline text-primary hover:text-underline block-link">Developers</a>
```
Comment (thisisdano) [inline pages/documentation/overview.md:18]: Point to Getting Started for Developers page

File: pages/documentation/overview.md [hunk 11]
```diff
@@ -15,15 +15,15 @@ redirect_from:
...
       </h3>
       <p class="margin-top-1">Get started with our code base by downloading the code as a zip file, or installing with npm.</p>
     </div>
   </div>
   <div class="margin-top-2 tablet:margin-top-0 tablet:grid-col display-flex flex-align-stretch">
     <div class="site-docs-card-link">
       <h3 class="font-lang-lg margin-0">
-        <a href="{{ site.baseurl }}/getting-started/designers/" class="text-no-underline text-primary hover:text-underline block-link">Designers</a>
+        <a href="{{ site.baseurl }}/documentation/getting-started-for-designers/" class="text-no-underline text-primary hover:text-underline block-link">Designers</a>
```
Comment (thisisdano) [inline pages/documentation/overview.md:26]: Point to Getting Started for Designers page

File: pages/documentation/getting-started-developers/phase-one.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+Now, the USWDS source code is in your project, in `./node_modules/uswds`.
+
+{: .site-note }
+**Don't modify the source code**
+This guidance may seem counterintuitive, but now that you've installed the USWDS source code, you don't want to edit it in any way because the source code in the `node_modules` directory is controlled by npm and could be rewritten at any time, especially if you reinstall or update the USWDS version. In fact, if you use Git or GitHub for your project, you should add `node_modules` to your `.gitignore` file.
+
+{: .site-note.margin-top-0 }
+You do want to use the source code, but you don't want to modify it to use it. Instead, in Phase 2, we'll discuss how to compile, theme, and extend the USWDS source code in your project.
+
+**Next:** [Phase 2: Compile &gt;]({{ site.baseurl }}/documentation/getting-started/developers/phase-two-compile/) →
```
Comment (mejiaj) [inline pages/documentation/getting-started-developers/phase-one.md:95]: Let's remove the `&gt;` now that we're using the arrow.

```suggestion
**Next:** [Phase 2: Compile]({{ site.baseurl }}/documentation/getting-started/developers/phase-two-compile/) →
```

**Review by mejiaj — CHANGES_REQUESTED** [review]
Just a small note on `phase-one.md`, otherwise LGTM.

File: pages/documentation/getting-started-developers/phase-three.md [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+For complete instructions on how to contribute code, please read [CONTRIBUTING.md](https://github.com/uswds/uswds/blob/develop/CONTRIBUTING.md). These instructions also include guidance on how to set up your own copy of the design system style guide website for development.
+
+If you would like to learn more about how we work, check out the [Workflow](https://github.com/uswds/uswds/wiki/Workflow) and [Issue label Glossary](https://github.com/uswds/uswds/wiki/Issue-label-glossary) pages on our wiki.
+
+If you have questions or concerns about our contributing workflow, feel free to [open an issue](https://github.com/uswds/uswds/issues), such as the following in GitHub:
+- Bug report — Report a bug and help USWDS improve.
+- Feature request — Suggest a new idea for the design system.
+- Report a security vulnerability — Report potential security issues. Review our [security policy](https://github.com/uswds/uswds/security/policy) for more details.
+
+You can also email us directly at [uswds@support.digitalgov.gov](mailto:uswds@support.digitalgov.gov).
```
Comment (mikki-stacey) [inline pages/documentation/getting-started-developers/phase-three.md:58]: Resources should be a separate section comprising all the content beneath it.

File: pages/documentation/getting-started-developers/overview.md [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+- npm
+- Sass (use [the latest version](https://nodejs.org/en/download/); we use [the latest `sass` package from npm](https://www.npmjs.com/package/sass))
+
+These instructions describe how to get started with USWDS using npm. If npm does not work for your project, you can [download and install the system directly](https://github.com/uswds/uswds#download-and-install) as outlined under “Download and install” in GitHub.
+
+We’ve designed USWDS to support older and newer browsers through [progressive enhancement](https://en.wikipedia.org/wiki/Progressive_enhancement). The design system follows the [2% rule](https://gds.blog.gov.uk/2012/01/25/support-for-browsers/): we officially support any browser above 2% usage as observed by [analytics.usa.gov](https://analytics.usa.gov/). In other words, USWDS version 2.0 supports Chrome, Firefox, Safari, Edge, and Internet Explorer 11 and up. The next major version of USWDS (3.0) will drop support of Internet Explorer 11.
+
+## Step-by-step
+Get started with USWDS by installing the design system on your project, compiling your project's Sass from the USWDS source, and customizing the design system to fit your project needs.
```
Comment (mikki-stacey) [inline pages/documentation/getting-started-developers/overview.md:43]:
```suggestion
Get started with USWDS by installing the design system on your project, compiling your project's Sass from the USWDS source, and customizing the design system to fit your project needs as outlined in the following phases:
```

**Pull request merged**